Modules/_datetimemodule.c

```diff
@@ -739,42 +811,60 @@ parse_hh_mm_ss_ff(const char *tstr, const char *tstr_end, int *hour,
     const char *p = tstr;
     const char *p_end = tstr_end;
     int *vals[3] = {hour, minute, second};
+    unsigned char has_separator = 1;
```

Comment:
Initializing this may confuse the reader, since you unconditionally overwrite this value below (and it's doing unnecessary throw-away work, though granted an optimizing compiler will probably optimize it out)
```suggestion
    unsigned char has_separator;
```

Reply:
This is done to avoid an erroneous compiler warning. I've added a comment.